tests/test_controller.py

```diff
@@ -2801,11 +2802,35 @@ def test_search(self):
         # Update the materialized view to make sure the works show up.
         SessionManager.refresh_materialized_views(self._db)
 
-        # Execute a search query designed to find the second one.
+        # Execute a search query designed to find the second work.
         with self.request_context_with_library("/?q=t&size=1&after=1"):
             # First, try the top-level lane.
             response = self.manager.opds_feeds.search(None)
+
             feed = feedparser.parse(response.data)
+
+            # When the feed links to itself or to another page of
+            # results, the arguments necessary to propagate the query
+            # string and facet information are propagated through the
+            # link.
+            def assert_propagates_facets(lane, link):
+                # Assert that the given `link` propagates
+                # the query string arguments found in the facets
+                # associated with this request.
+                facets = load_facets_from_request(
+                    self._default_library, lane, base_class=SearchFacets
+                )
+                for k, v in facets.items():
+                    check = '%s=%s' % tuple(map(urllib.quote, (k,v)))
+                    assert check in link['href']
+
+            feed_links = feed['feed']['links']
+            for rel in ('next', 'previous', 'self'):
+                [link] = [link for link in feed_links if link.rel == rel]
+
+                assert_propagates_facets(None, link)
+                assert 'q=t' in link['href']
+
```

Comment:
I screwed up the code that this is testing and then noticed that it wasn't being tested, so I added some coverage.